### DISCUSSION TITLE
How to style child element depending on data attribute of parent (open & closed)?

### DISCUSSION THREAD

**Discussion by f3bw** [tailwindlabs/tailwindcss#10301] [Help] [answered]
Hi there,

I want to give a child element styling depending on the data state (open or closed) of the parent element. How do I achieve this with TailwindCSS?

### Example:

`<Parent
        className="data-[state=open]:[&>svg]:rotate-180"
      >
        <ChildIcon className="transition" />
      </Parent>`

**Comment by wongjn** [comment, marked as answer]
In this instance, the variant order is important – [they are applied inside-out](https://tailwindcss.com/docs/hover-focus-and-other-states#ordering-stacked-modifiers) so you'd want something like:
```html
<Parent className="[&>svg]:data-[state=open]:rotate-180">
```
You could also consider using [`group-*`](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state) like:
```html
<Parent className="group">
  <svg className="group-data-[state=open]:rotate-180">
```

**Reply by f3bw** [reply]
Thank you for the explanation, I appreciate it!

**Reply by 7iomka** [reply]
Hi. How to make the className when group has state open and is hovered at the same time?

**Reply by wongjn** [reply]
- `group-data-[state=open]:hover:` for when the element itself is hovered.
- `group-data-[state=open]:group-hover:` for when the `group` element is hovered.
